helix-core/src/test/java/org/apache/helix/integration/task/TestExecutionDelay.java

```diff
@@ -0,0 +1,124 @@
+
+/**
+ * Test To check if the Execution Delay is respected.
+ */
+public class TestExecutionDelay extends TaskTestBase {
+
+  @BeforeClass
+  public void beforeClass() throws Exception {
+    super.beforeClass();
+  }
+
+  @Test
+  public void testExecutionDelay() throws Exception {
+    // Execution Delay is set to be a large number. Hence, the workflow should not be completed
+    // soon.
+    final long executionDelay = 100000000000L;
+    String workflowName = TestHelper.getTestMethodName();
+    Workflow.Builder builder = new Workflow.Builder(workflowName);
+
+    // Workflow DAG Schematic:
+    //          JOB0
+    //           /\
+    //          /  \
+    //         /    \
+    //       JOB1   JOB2
+
+    JobConfig.Builder jobBuilder = JobConfig.Builder.fromMap(WorkflowGenerator.DEFAULT_JOB_CONFIG)
+        .setMaxAttemptsPerTask(1).setWorkflow(workflowName)
+        .setJobCommandConfigMap(ImmutableMap.of(MockTask.JOB_DELAY, "1000"));
+
+    JobConfig.Builder jobBuilder2 = JobConfig.Builder.fromMap(WorkflowGenerator.DEFAULT_JOB_CONFIG)
+        .setMaxAttemptsPerTask(1).setWorkflow(workflowName)
+        .setJobCommandConfigMap(ImmutableMap.of(MockTask.JOB_DELAY, "1000"));
+
+    JobConfig.Builder jobBuilder3 = JobConfig.Builder.fromMap(WorkflowGenerator.DEFAULT_JOB_CONFIG)
+        .setMaxAttemptsPerTask(1).setWorkflow(workflowName)
+        .setJobCommandConfigMap(ImmutableMap.of(MockTask.JOB_DELAY, "1000"));
+
+    builder.addParentChildDependency("JOB0", "JOB1");
+    builder.addParentChildDependency("JOB0", "JOB2");
+    builder.addJob("JOB0", jobBuilder.setExecutionDelay(executionDelay));
+    builder.addJob("JOB1", jobBuilder2);
+    builder.addJob("JOB2", jobBuilder3);
+
+    // Since the execution delay is set as a long time, it is expected to reach time out.
+    // If the workflow completed without any exception, it means that ExecutionDelay has not been
+    // respected.
+    _driver.start(builder.build());
+
+    // Verify workflow Context is created.
+    // Wait until Workflow Context is created.
+    boolean resultWorkflow = TestHelper.verify(() -> {
+      WorkflowContext wCtx1 = _driver.getWorkflowContext(workflowName);
+      return (wCtx1 != null);
+    }, 60 * 1000);
+    Assert.assertTrue(resultWorkflow);
+
+    // Verify StartTime is set for the Job.
+    // Wait until startTime is set.
+    boolean resultStartTime = TestHelper.verify(() -> {
+      WorkflowContext wCtx1 = _driver.getWorkflowContext(workflowName);
+      long startTime = -1;
+      try {
+        startTime = wCtx1.getJobStartTime(TaskUtil.getNamespacedJobName(workflowName, "JOB0"));
+      } catch (Exception e) {
+        // pass
+      }
+      return (startTime != -1);
+    }, 60 * 1000);
+    Assert.assertTrue(resultStartTime);
+
+    // Verify Job Context is not created. If Job Context is created, it means that the execution
+    // delay is not respected. if not created, it means that the execution delay is respected.
```

Comment:
is respected because the absence of JobContext implies that the job did not run.